src/tests/unit/sync/syncimages_ring.f90

```diff
@@ -0,0 +1,25 @@
+! SYNC IMAGES([this_image - 1, this_image + 1]) with the STAT=STAT_STOPPED_IMAGE
+! specifier and wrap around of image numbers.  The test is intended to check
+! that syncing in a ring with a stopped image still terminates all images.
+
+program sync_images_ring
+  use, intrinsic:: iso_fortran_env
+  implicit none
+
+  integer :: stat_var = 0
+
+  if (num_images() .lt. 2) error stop "Need at least two images to test."
+
+  associate (me => this_image())
+    if (me /= 1) then
```

Comment:
I'm not 100% confident in the semantics of calling `stop` (**NOT** `error stop`) from a coarray program, but if you can call `stop 0` from image `me == 1` explicitly without causing all the other images to abort, that would clarify the test.